### PR TITLE
Readme - HAL structure

### PR DESCRIPTION
**Pull request by 0xc0170**
Resuming #7 as I consider this vital for this repository. This should a good base for porting, or simply understanding the hal hierarchy.

I added target module to the picture, plus fixed some formatting.

@sg- @bremoran @bogdanm @rgrover

### CODE REVIEW

**Comment by bogdanm** [comment]
+1 from me, nice work! We needed this.

File: readme.md [hunk 1]
```diff
@@ -15,4 +15,102 @@ TBC
...
+
+#### mbed-hal-\<vendor\>-\<target\>
+The target-specific definitions expected by the vendor HAL and/or mbed-hal-\<vendor\>-\<chip family\>
+
+Contains:
+* source
+  * init_api.c
+  * PeripheralPins.c
+* mbed-hal-\<vendor\>-\<target\>
+  * device.h
```
Comment (rgrover) [inline readme.md]: isn't there a objects.h and gpio_object.h also needed?

**Comment by sg-** [comment]
Great start but its a bit confusing. Its a partial porting guide and partial document. I think these readmes should be scoped to the repo only and if there is a program for porting or just a repo with submodules that pull all the <vendor> modules that is where the porting guide would live.

File: readme.md (hunk 1 above)
Comment (0xc0170) [inline readme.md]: objects.h are currently referenced in the above layers, at least in minar, which we can remove, so objects could die and gpio_object.h should not be so then we can remove it. will update once I do proper grep

**Comment by 0xc0170** [comment]
>  Its a partial porting guide and partial document.

@sg- it is because there's no porting guide. Once we have one, I imagine this would have less detail (something to be defined later what) and link to a porting guide. If you got better idea, shoot here now. This mbed hal needs some more details than it contains, and I am here to write it :)

**Comment by 0xc0170** [comment]
@sg- I'll merge this as it.cPlease send edits, or create an issue regarding what should be in this readme.

**Comment by 0xc0170** [comment]
Created tracking issue referenced above for "Add a link to a porting guide and do future edits"

**Pull request merged**